File: no-auth/src/main/java/com/kjj/noauth/controller/TestController.java
package com.kjj.noauth.controller;

import com.kjj.noauth.service.UserService;
import com.kjj.noauth.util.jwt.JwtTools;
import lombok.RequiredArgsConstructor;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

import java.util.Map;

@RestController
@RequestMapping({"/v1/no-auth"})
@RequiredArgsConstructor
public class TestController {

    private final JwtTools jwtTools;
    private final UserService userService;

    @GetMapping("test")
    public ResponseEntity<Map<String, String>> testToken(@RequestParam("username") String username){
        if (username == null) username = "manager";
        return ResponseEntity.ok(Map.of("accessToken", jwtTools.createToken(userService.loadUserByUsername(username))));
    }
}

File: no-auth/src/main/kotlin/com/kjj/noauth/controller/TestController.kt
package com.kjj.noauth.controller

import com.kjj.noauth.service.UserService
import com.kjj.noauth.tool.jwt.JwtTool
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RequestParam
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/v1/no-auth")
class TestController(
    private val jwtTool: JwtTool,
    private val userService: UserService
) {
    @GetMapping("test")
    fun testToken(@RequestParam("username") username: String?): ResponseEntity<Map<String, String>> {
        val finalUsername = username ?: "manager"
        return ResponseEntity.ok(
            java.util.Map.of(
                "accessToken",
                jwtTool.createToken(userService.loadUserByUsername(finalUsername))
            )
        )
    }
}